Fixed day calendar view to display correct local preference

boxlady opened the pull request:
Changed Calendar day view to display the correct time format based on global setting, preference/ local (24 or 12 hour). Other views (week etc.)  need to be changed too, but I thought I would just try to push this one first.

below added by bradymiller:
Up For Grabs demo for this PR is here:
https://www.open-emr.org/wiki/index.php/Development_Demo#Delta_-_Up_For_Grabs_Demo

bradymiller commented:
Up For Grabs demo for this PR is here:
https://www.open-emr.org/wiki/index.php/Development_Demo#Delta_-_Up_For_Grabs_Demo

bradymiller commented:
hi @boxlady ,
I set up a demo, so folks can test this out. Your coding approach looks good and testing looks good. Recommend bringing in the changes on the week and month screens into this PR also. thanks! -brady

boxlady commented:
> hi @boxlady , see how this is a user specific global, https://github.com/openemr/openemr/blob/master/library/globals.inc.php#L130
> 
> looks like this keeps the current functionality, are you trying to change it so that you display 24 hr times on the calendar? thank you

Yes, along with some other places. Here in europe they use 24 hours for everything and that is their preference. I noticed that changing the time format in globles/local didnt do anything I I thought about setting out to fix that.

bradymiller commented on interface/main/calendar/add_edit_event.php:
```diff
@@ -81,7 +87,7 @@
     $starttimeh = $_GET['starttimeh'];
     if (isset($_GET['startampm'])) {
-        if ($_GET['startampm'] == '2' && $starttimeh < 12) {
+        if ($_GET['startampm'] == '2' && $starttimeh < 12 && $globle_timeformat == 1) {
```
Wouldn't it be simpler to just use $GLOBALS['time_display_format'] here and drop the code block at the top?

boxlady replied:
Start time hour is set at the top, so I changed it at the bottom for simplicity.

bradymiller commented on interface/main/calendar/add_edit_event.php:
```diff
@@ -1010,7 +1016,7 @@ function setEventDate($start_date, $recurrence)
     if ($starttimeh >= 12) { // p.m. starts at noon and not 12:01
         $startampm = '2';
-        if ($starttimeh > 12) {
+        if ($starttimeh > 12 && $globle_timeformat == 1) {
```
Wouldn't it be simpler to just use $GLOBALS['time_display_format'] here and drop the code block at the top?

bradymiller replied:
Recommend using the $GLOBALS['time_display_format'] directly.

bradymiller commented on interface/main/calendar/modules/PostCalendar/pntemplates/default/views/day/ajax_template.html:
```diff
@@ -411,10 +412,15 @@
 //============================================================================================================================
+// Check global time preference so that 24 and 12hour preference is displayed properly
+$timeformat =0;
+if( $globle_timeformat == 1){ $timeformat = 12; }
```
Wouldn't it be simpler to just use $GLOBALS['time_display_format'] here and drop the line of code at the top?

bradymiller commented on interface/main/calendar/modules/PostCalendar/pntemplates/default/views/day_print/outlook_ajax_template.html:
```diff
@@ -287,10 +287,13 @@
 echo "<td id='times' style='width:2em;'><div><table>\n";
 echo "<td class='timeslot' style='text-align: center; background-color:#ddd;'>&nbsp;</td>";
+$timeformat= 0;
+if ($GLOBALS['time_display_format'] == 1){   $timeformat = 12; }
```
Wouldn't it be simpler to just use $GLOBALS['time_display_format'] here?

bradymiller commented on interface/main/calendar/modules/PostCalendar/pntemplates/default/views/week/ajax_template.html:
```diff
@@ -454,10 +454,13 @@
     echo "<td id='times'><div><table>\n";
 //==================================================================================================================
+$timeformat =0;
+ if( $globle_timeformat == 1){ $timeformat = 12; }
```
Wouldn't it be simpler to just use $GLOBALS['time_display_format'] here?

bradymiller commented on interface/main/calendar/add_edit_event.php:
```diff
@@ -81,7 +87,7 @@
 if (isset($_GET['starttimeh'])) {
```
In this script, it looks like $starttimeh feeds the form_hour form element. Are we sure this doesn't break anything when form is submitted and doesn't break any of the related javascript stuff in this script?

boxlady replied:
I made some changes to this for the am/pm fix.  From what I can see it doesnt break it it, however I'm not the best with javascript.

bradymiller commented on interface/patient_file/summary/demographics.php:
```diff
@@ -1673,7 +1673,7 @@ function setMyPatient() {
                             $dispmin  = substr($row['pc_startTime'], 3, 2);
                             if ($disphour >= 12) {
```
In this script, it looks like may also need to deal with the $dispampm variable (should it not be either am or pm when using 24H time?).

boxlady replied:
It still sets it based on whether it is greater than 12. I could remove it but I wasn't sure  if it was better to leave it so it says 13:45 pm or to remove it altogether.

bradymiller replied:
remove the am/pm when using 24H time

bradymiller commented:
hi @boxlady , I did another code review and left some more comments. Plan to do more testing when these are addressed. (also, don't worry to much about the travis stuff; I can fix this up via a script when it's ready for the codebase) -brady

bradymiller commented:
hi @boxlady , Just let me know when your ready for me to do another code review/test.

boxlady commented:
@Brady Miller - I think it's ready for test/review. Do you think I should
get rid of am/pm in demographics?


On Thu, Jul 25, 2019 at 7:53 AM Brady Miller <notifications@github.com>
wrote:

> hi @boxlady <https://github.com/boxlady> , Just let me know when your
> ready for me to do another code review/test.
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/openemr/openemr/pull/2568?email_source=notifications&email_token=ABHAJ5M5CETMF5K4KZUEZL3QBE5XBA5CNFSM4IEYVRH2YY3PNVWWK3TUL52HS4DFVREXG43VMVBW63LNMVXHJKTDN5WW2ZLOORPWSZGOD2YNY3I#issuecomment-514907245>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/ABHAJ5PVD6AGZKTC7NKWLB3QBE5XBANCNFSM4IEYVRHQ>
> .
>


-- 
Julie Buurman

bradymiller commented:
hi @boxlady ,
Lets remove the am/pm when using 24H time:
https://ell.stackexchange.com/questions/132685/should-you-add-am-or-pm-after-a-24-hour-time

bradymiller commented on interface/main/calendar/add_edit_event.php:
```diff
@@ -64,6 +64,12 @@
+$timeformat = 0;
+if ($GLOBALS['time_display_format'] == 1) {
+    $timeformat = 12;
+}
```
The above code is not used below in the script.

bradymiller replied:
Above code block is not used in this script, so should be removed

bradymiller replied:
hi @boxlady , Just making sure you saw the post above:
```
Above code block is not used in this script, so should be removed
```

bradymiller commented on interface/main/calendar/add_edit_event.php:
```diff
@@ -80,7 +86,7 @@
 if (isset($_GET['starttimeh'])) {
     $starttimeh = $_GET['starttimeh'];
-    if (isset($_GET['startampm'])) {
+    if ($globle_timeformat == 1 && isset($_GET['startampm'])) {
```
Recommend using the $GLOBALS['time_display_format'] directly.

bradymiller commented on interface/main/calendar/add_edit_event.php:
```diff
@@ -1474,10 +1480,12 @@ function find_available(extra) {
-                <option value='1'><?php echo xlt('AM'); ?></option>
-                <option value='2'<?php echo ($startampm == '2') ? " selected" : ""; ?>><?php echo xlt('PM'); ?></option>
-            </select>
+            <?php if ($globle_timeformat == 1): ?>
```
Recommend using the $GLOBALS['time_display_format'] directly.

bradymiller commented on interface/main/calendar/add_edit_event.php:
```diff
@@ -2155,7 +2163,8 @@ function SubmitForm() {
         // Check slot availability.
         var mins = parseInt(f.form_hour.value) * 60 + parseInt(f.form_minute.value);
-        if (f.form_ampm.value == '2' && mins < 720) mins += 720;
+        <?php if($globle_timeformat == 1):?>if (f.form_ampm.value == '2' && mins < 720) mins += 720;<?php endif ?>
```
Recommend using the $GLOBALS['time_display_format'] directly.

bradymiller commented on interface/main/calendar/modules/PostCalendar/pntemplates/default/views/day/ajax_template.html:
```diff
@@ -99,6 +99,7 @@
 
   $openhour= $GLOBALS['schedule_start'];
   $closehour= $GLOBALS['schedule_end'];
+  $globle_timeformat= $GLOBALS['time_display_format'];
```
recommend removing above line

bradymiller commented on interface/main/calendar/modules/PostCalendar/pntemplates/default/views/day/ajax_template.html:
```diff
@@ -411,10 +412,17 @@
 //============================================================================================================================
+// Check global time preference so that 24 and 12hour preference is displayed properly
+
+if( $globle_timeformat == 1){
```
Recommend using the $GLOBALS['time_display_format'] directly.

bradymiller commented on interface/main/calendar/modules/PostCalendar/pntemplates/default/views/day/ajax_template.html:
```diff
@@ -411,10 +412,17 @@
+
+if( $globle_timeformat == 1){
+   $timeformat = 12;
+   } else { $timeformat =0; }
```
would be better to format as:
```
if ($GLOBALS['time_display_format'] == 1){
    $timeformat = 12;
} else {
    $timeformat = 0;
}
```

bradymiller commented on interface/main/calendar/modules/PostCalendar/pntemplates/default/views/day_print/outlook_ajax_template.html:
```diff
@@ -287,10 +287,16 @@
+
+if( $globle_timeformat == 1){
+  $timeformat = 12;
+} else { $timeformat =0; }
```
```
if ($GLOBALS['time_display_format'] == 1) {
    $timeformat = 12;
} else {
    $timeformat = 0;
}
```

bradymiller commented on interface/main/calendar/modules/PostCalendar/pntemplates/default/views/week/ajax_template.html:
```diff
@@ -454,10 +454,15 @@
 //==================================================================================================================
+if( $globle_timeformat == 1){
+   $timeformat = 12;
+} else { $timeformat =0; }
```
```
if ($GLOBALS['time_display_format'] == 1) {
    $timeformat = 12;
} else {
    $timeformat = 0;
}
```

bradymiller commented:
hi @boxlady , Completed review. Let me know when ready for another review. thanks! -brady

boxlady commented:
@bradymiller  Made the changes.  Ready for another review when you have time. Thank you :)

boxlady commented:
Can this not be added?

bradymiller commented:
hi @boxlady , Sorry about the serious delay. We just implemented a stale bot on github to help us remove outdated issues/PRs (and identify ones that we forgot about like this one). I'll add this to my list of PR's to review to ensure I review this again in near future.

boxlady commented:
Updated to master, fixed conflicts. For most EU Countries they use military time (24 hours). So it would be handy if this could be put in to master.

bradymiller commented on interface/main/calendar/add_edit_event.php:
```diff
@@ -46,6 +46,7 @@
 require_once($GLOBALS['srcdir'] . '/calendar.inc');
 require_once($GLOBALS['srcdir'] . '/options.inc.php');
 require_once($GLOBALS['srcdir'] . '/encounter_events.inc.php');
+require_once($GLOBALS['srcdir'].'/acl.inc');
```
this file no longer exists, so line above should be removed

bradymiller replied:
unable to test this script since get whitescreen of death from above line

bradymiller commented:
hi @boxlady , Just did a review on this. Testing was limited secondary to the error. Agree we need to get this into the codebase and look forward to the next revision where can do more testing (and likely get it into codebase at that time). thanks! -brady

boxlady commented:
> hi @boxlady , Just did a review on this. Testing was limited secondary to the error. Agree we need to get this into the codebase and look forward to the next revision where can do more testing (and likely get it into codebase at that time). thanks! -brady

Hi @bradymiller - Sorry I missed that forgot that it was changed over. It should work now. Thanks for the quick reply...

boxlady commented:
I didn't but I'll fix it tomorrow.

On Sat, 6 Jun 2020, 9:50 am Brady Miller, <notifications@github.com> wrote:

> *@bradymiller* commented on this pull request.
> ------------------------------
>
> In interface/main/calendar/add_edit_event.php
> <https://github.com/openemr/openemr/pull/2568#discussion_r436248390>:
>
> > @@ -64,6 +64,12 @@
>
>   $_POST['form_date'] = DateToYYYYMMDD($_POST['form_date']);
>   $_POST['form_enddate'] = DateToYYYYMMDD($_POST['form_enddate']);
> +// Check preferences for time format
> +$timeformat = 0;
> +if ($GLOBALS['time_display_format'] == 1) {
> +    $timeformat = 12;
> +}
>
> hi @boxlady <https://github.com/boxlady> , Just making sure you saw the
> post above:
>
> Above code block is not used in this script, so should be removed
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/openemr/openemr/pull/2568#discussion_r436248390>, or
> unsubscribe
> <https://github.com/notifications/unsubscribe-auth/ABHAJ5NJXQFQCZUXMYWYZLLRVHYNHANCNFSM4IEYVRHQ>
> .
>

bradymiller commented:
hi @boxlady , Code looks good. Gonna bring this in and then fix some minor psr2 stuff. Thanks for the improvements! -brady
